spa_ui/self_service/client/app/states/marketplace/list/list.state.js

```diff
@@ -110,11 +128,13 @@
     }
 
     function matchesFilter(item, filter) {
-      var match = true;
+      var match = false;
       if (filter.id === 'template_name') {
         match = (String(item.name).toLowerCase().indexOf(String(filter.value).toLowerCase()) > -1 );
       } else if (filter.id === 'template_description') {
         match = (String(item.long_description).toLowerCase().indexOf(String(filter.value).toLowerCase()) > -1 );
+      } else if (filter.id === 'catalog_name' && typeof item.service_template_catalog !== 'undefined') {
```

Comment:
I imagine null might be undesired here as well, possible to shorthand to `&& typeof item.service_template_catalog` ?